### PR TITLE
WIP: Reworking of SLOSH and Meta-classifier

### PR DESCRIPTION
**Pull request by jsk389**
These are the changes that I have made to SLOSH and the meta-classifier, which should result in it training better! I need to clean up the code a little and remove bits here and there that shouldn't be there.

### CODE REVIEW

**Comment by jsk389** [comment]
I've cleaned up the code, so hopefully it should be in a position where everything should run without issues.

File: starclass/training_sets/training_set.py [hunk 1]
```diff
@@ -47,15 +50,16 @@ def __init__(self, datalevel='corr', tf=0.0):
...
 				test_size=self.testfraction,
-				random_state=42,
-				stratify=self.labels()
+				random_state=self.random_seed,
+				stratify=labels, #self.labels()
 			)
-			# Have to sort as train_test_split shuffles and we don't want that
-			self.train_idx = np.sort(self.train_idx)
-			self.test_idx = np.sort(self.test_idx)
+		# Have to sort as train_test_split shuffles and we don't want that
```
Comment (rhandberg) [inline starclass/training_sets/training_set.py]: No reason to change these three lines... It will be sorted if coming directly from `arange`

Comment (jsk389) [reply starclass/training_sets/training_set.py]: That's true, I put it outside just to make doubly sure that everything is sorted.  `train_test_split` shuffles the data so it needs to be resorted otherwise the wrong labels are picked out later on, so it definitely needs to be sorted inside if `test_fraction != 0`. I will change this now.

Comment (jsk389) [reply starclass/training_sets/training_set.py]: This should now have been sorted

File: starclass/training_sets/training_set.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
 #----------------------------------------------------------------------------------------------
 class TrainingSet(object):
 
-	def __init__(self, datalevel='corr', tf=0.0):
+	def __init__(self, datalevel='corr', tf=0.0, random_seed=42):
```
[the remark this answers is not in the document]
Comment (jsk389) [reply starclass/training_sets/training_set.py]: Definitely! I just did this to ensure everything I did was reproducible, but setting it to be random by default is definitely the way we should do it.

**Comment by rhandberg** [comment]
Merged into devel... I will fix those couple of lines after the merge
